use_cluster.py

```diff
@@ -366,4 +355,391 @@ def return_cluster_dict(q_cluster, q_ids_list):
         else:
             final_clusters[q_cluster[indx]].append(q_ids_list[indx])
 
-    return final_clusters
\ No newline at end of file
+    return final_clusters
+
+
+"""
+> The objective of this function is to generate labels for each cluster based on a custom score developed by Araz, 
+using Normalised Mutual Information (NMI) and Distance from the Cluster Centroid 
+:The inputs required are the embeddings (generated previously from cluster functions), list of questions, question ids & clusters generated 
+: Read about NMI here: https://arxiv.org/pdf/1702.08199.pdf
+: For how Araz developed the custom score/metric, refer here: 
+"""
+
+def return_cluster_labels_NMI_nGrams_Centroid(embeddings,qs_list,q_ids_list,clusters): 
+
+    labelling_corpus = LabellingClusterCorpus(qs_list, nlp)
+    labelling_corpus.process_docs(embeddings, q_ids_list, clusters)
+    
+    
+    cluster_objs, vecs, global_keywords, lemmatized_qs_list = Generate_Modified_Qs_Data(labelling_corpus.documents, clusters, q_ids_list,1)
```

Comment:
I'm not really sure why global_keywords and vecs are computed inside Generate_Modified_Qs_Data. It seems like it'd be simpler just to add lines to compute them directly here. In general, I would avoid functions that return several parameters where possible - if the pieces can be computed independently of each other, it makes more sense to have a separate function to compute each one. That way the functions are all independently reusable.

Reply:
Yes agreed Jeff! Initially, when I didn't have a separate function for Global Keywords, they were being calculated in this only, so afterward I didn't realize that it could be called independently. Have called it in the parent function separately now :D